src/mtev_http.c

```diff
@@ -981,6 +1008,13 @@ mtev_http_complete_request(mtev_http_session_ctx *ctx, int mask) {
     if(len <= 0) goto full_error;
     if(len > 0) in->size += len;
     rv = mtev_http_request_finalize(&ctx->req, &err);
+    /* walk the bchain and set the compression */
```

Comment:
this will "work", but is inefficient - we'll enter this routine multiple times while accumulating headers, and the `bchain` isn't fully formed until the last time through. It'd be better to walk the bchain only if `rv == mtev_true`, AFAICT.